diff --git a/node_modules/keycloakify/src/bin/keycloakify/generateFtl/generateFtl.ts b/node_modules/keycloakify/src/bin/keycloakify/generateFtl/generateFtl.ts
old mode 100644
new mode 100755
diff --git a/node_modules/keycloakify/src/bin/keycloakify/generateFtl/index.ts b/node_modules/keycloakify/src/bin/keycloakify/generateFtl/index.ts
old mode 100644
new mode 100755
diff --git a/node_modules/keycloakify/src/bin/keycloakify/generateFtl/kcContextDeclarationTemplate.ftl b/node_modules/keycloakify/src/bin/keycloakify/generateFtl/kcContextDeclarationTemplate.ftl
old mode 100644
new mode 100755
index 1dce06d..27022b3
--- a/node_modules/keycloakify/src/bin/keycloakify/generateFtl/kcContextDeclarationTemplate.ftl
+++ b/node_modules/keycloakify/src/bin/keycloakify/generateFtl/kcContextDeclarationTemplate.ftl
@@ -260,8 +260,7 @@ function decodeHtmlEntities(htmlStr){
                     key == "entity" &&
                     areSamePath(path, ["user"])
                 ) || (
-                    key == "attributes" &&
-                    areSamePath(path, ["realm"])
+                    areSamePath(path, ["realm", "attributes"]) && !key?starts_with("custom")
                 ) || (
                     xKeycloakify.pageId == "index.ftl" &&
                     xKeycloakify.themeType == "account" &&
